src/components/dialogs/parameters/sensi/sensitivity-analysis-parameters.js

```diff
@@ -0,0 +1,441 @@
+/**
+ * Copyright (c) 2023, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+import { SubmitButton, useSnackMessage } from '@gridsuite/commons-ui';
+import { Grid, Button, DialogActions } from '@mui/material';
+import { useCallback, useEffect, useMemo, useState } from 'react';
+import { FormattedMessage } from 'react-intl';
+import { useSelector } from 'react-redux';
+import { useStyles, CloseButton, DropDown } from '../parameters';
+import { yupResolver } from '@hookform/resolvers/yup';
+import { FormProvider, useForm } from 'react-hook-form';
+import {
+    ANGLE_FLOW_SENSITIVITY_VALUE_THRESHOLD,
+    CONTINGENCIES,
+    EQUIPMENTS_IN_VOLTAGE_REGULATION,
+    FILTER_ID,
+    FILTER_NAME,
+    FLOW_FLOW_SENSITIVITY_VALUE_THRESHOLD,
+    FLOW_VOLTAGE_SENSITIVITY_VALUE_THRESHOLD,
+    HVDC_LINES,
+    ID,
+    DISTRIBUTION_TYPE,
+    INJECTIONS,
+    MONITORED_BRANCHES,
+    NAME,
+    PARAMETER_SENSI_HVDC,
+    PARAMETER_SENSI_INJECTION,
+    PARAMETER_SENSI_INJECTIONS_SET,
+    PARAMETER_SENSI_NODES,
+    PARAMETER_SENSI_PST,
+    PSTS,
+    SENSITIVITY_TYPE,
+    SUPERVISED_VOLTAGE_LEVELS,
+} from '../../../utils/field-constants';
+import yup from '../../../utils/yup-config';
+import {
+    getSensitivityAnalysisParameters,
+    setSensitivityAnalysisParameters,
+} from '../../../../services/study/sensitivity-analysis';
+import SensitivityAnalysisFields from './sensitivity-Flow-parameters';
+import {
+    getSensiInjectionsSetformatNewParams,
+    getSensiInjectionsSetFormSchema,
+} from './sensi-injections-set';
+import {
+    getSensiInjectionsformatNewParams,
+    getSensiInjectionsFormSchema,
+} from './sensi-injections';
+import {
+    getSensiHvdcformatNewParams,
+    getSensiHVDCsFormSchema,
+} from './sensi-hvdcs';
+import {
+    getSensiPstformatNewParams,
+    getSensiPSTsFormSchema,
+} from './sensi-psts';
+import {
+    getSensiNodesformatNewParams,
+    getSensiNodesFormSchema,
+} from './sensi-nodes';
+import SensiParametersSelector from './sensi-parameters-selector';
+import { LineSeparator } from '../../dialogUtils';
+
+export const useGetSensitivityAnalysisParameters = () => {
```

Comment:
This custom hook have to be on a seperate file. There is a file named custom-hooks. We can use it